wurst/network/Network.wurst

```diff
@@ -0,0 +1,922 @@
+package Network
+
+import SyncSimple
+import Table
+import ErrorHandling
+import MoriUtil
+
+// max amount to be written/read/synced per an execution to avoid hitting the OP limit
+// this value can be higher if Wurst is compiled without stack traces and with all optimizations turned on
+// and even higher with safety checks off
+@configurable constant integer DATA_PER_EXECUTE = 256
+// max length of a GC key
+// you may fine-tune it as you see fit to see what works faster
+@configurable constant integer MAX_CHARACTERS_PER_KEY = 3
+// setting this to false will optimize out all safety checks...
+// only disable this if you are 100% sure your code works correctly
+@configurable constant boolean SAFETY_CHECKS_ENABLED = true
+
+//TODO: I just realized that we can eliminate some delay by sending metadata and the first round together
+// and avoid one round-trip, essentially reducing transmission time for small payloads up to two times
+
+/********************************************************************************
+	Network library v1.0 by MoriMori.
+
+	Based on TriggerHappy's Sync library.
+
+	Rationale:
+		This library can be used to send arbitrary amounts of integer, real, boolean
+		and string data from one player to the rest.
+
+		Like SyncSimple, it abuses the fact that all network actions in WC3 are
+		sequential, and are received by players in the same order that are sent in.
+
+		It uses the SyncStored* natives of the gamecache to send data to other players
+		from the sender, and then uses SyncSimple to send a 'finish' event, that will
+		only be received by other players after they have also received all sync data.
+
+		The Network class provides 4 independent streams of data:
+			integers,
+			reals,
+			booleans,
+			strings
+
+		each of which can be written/read to using the relevant 
+		methods of the DataStream class.
+
+		Before sending the data, the sender populates the DataStream with the data that
+		they want to send, then Network.start() is called, as well as Network.onFinish(callback)
+		and data is received from the same buffer inside the callback when it has all been received.
+
+		Read SyncSimple docs for a slightly more in-depth overview of this particular
+		peculiarity of WC3.
+
+	Usage:
+		1. Create a new Network object using:
+			// sender is the player to receive data from
+			let network = new Network(sender)
+
+		2. Queue some data for sending from the sender. You can write as much as you need, there is no limit:
+			if localPlayer == sender
+				// keep in mind that before you start sending data,
+				// network.getData() is locked in a WRITE-ONLY MODE
```

Comment:
Why? What does that mean?

Reply:
Hopefully makes much more sense with file example explanation.

Reply:
Sorry, the purpose of my comment was to indicate that your code comment wasn't clear enough.

Rule of thumb: code comments explain why, not what.